src/sys/socket/mod.rs

```diff
@@ -517,16 +538,18 @@ pub fn recvmsg<'a, T>(fd: RawFd, iov: &[IoVec<&mut [u8]>], cmsg_buffer: Option<&
         Some(cmsg_buffer) => (cmsg_buffer as *mut _, mem::size_of_val(cmsg_buffer)),
         None => (0 as *mut _, 0),
     };
-    let mut mhdr = msghdr {
-        msg_name: &mut address as *const _ as *const c_void,
-        msg_namelen: mem::size_of::<sockaddr_storage>() as socklen_t,
-        msg_iov: iov.as_ptr() as *const IoVec<&[u8]>, // safe cast to add const-ness
-        msg_iovlen: iov.len() as type_of_msg_iovlen,
-        msg_control: msg_control as *const c_void,
-        msg_controllen: msg_controllen as type_of_cmsg_len,
-        msg_flags: 0,
+    let mut mhdr = unsafe {
+        let mut mhdr: msghdr = mem::uninitialized();
```

Comment:
Shouldn't think be `mem::zero()` so we make sure all memory is in a known state?

Reply:
I'm fine with making this change in all of the places suggested, but the prevailing style seems to be towards uninitialized() and zeroed() potentially has a small additional cost.

Reply:
I was just worried that something might not have gotten initialized. But given that we don't access any of the uninitialized fields, and there's no way for this datatype to escape our wrapper, this makes sense.